src/precomputed_quantities.jl

```diff
@@ -15,10 +16,24 @@ end
 function precomputed_quantities!(Y, p, t, colidx)
     ᶜuₕ = Y.c.uₕ
     ᶠw = Y.f.w
-    (; ᶜK, ᶠu_tilde, ᶜu_bar, ᶠu³, ᶜts, ᶜp, params) = p
-    (; ᶠwinterp, ᶜinterp) = p.operators
+    (;ᶠuₕ_bc, u₃_bc, ᶜK, ᶠu_tilde, ᶜu_bar, ᶠu³, ᶜts, ᶜp, g¹³, g²³, g³³, params) = p
+    (; ᶠinterp, ᶠwinterp, ᶜinterp) = p.operators
     ᶜJ = Fields.local_geometry_field(axes(Y.c.ρ)).J
     C123 = Geometry.Covariant123Vector
+    ᶠuₕ_bc[colidx] .=
+        Geometry.Covariant12Vector.(
+            Fields.level(ᶠinterp.(ᶜuₕ[colidx]), Utilities.half)
+        )
+    u₃_bc[colidx] .= @. Geometry.Covariant3Vector(
+        -1 * (
+            g¹³[colidx] * ᶠuₕ_bc.components.data.:1[colidx] +
+            g²³[colidx] * ᶠuₕ_bc.components.data.:2[colidx]
+        ) / g³³[colidx],
+    )
+    enforce_boundary = Operators.SetBoundaryOperator(
+        bottom = Operators.SetValue(u₃_bc[colidx]),
+    )
+    @. ᶠw[colidx] = enforce_boundary(ᶠw[colidx])
```

Comment:
I don't think this is right: to be consistent, we want to define it so that $\bar{u}^3 = 0$. 

1) interpolate the 3rd contravariant component of $u_h$ to he boundary: since $I(\rho J)$  at the boundary is equal to $\rho J$ at the first lecel, we can just use regular (unweighted) interpolation:
```julia
@. Fields.level(ᶠinterp(Geometry.contravariant3(ᶜuₕ[colidx])), half)
```
2) divide this by the 3rd contravariant component of a unit `Covariant3Vector` at the boundary:
```julia
@. Fields.level(Geometry.contravariant3(one(ᶠw[colidx])), half)
```

3) Set the boundary to velocity to be `- (1) / (2)`. Putting this altogether:
```julia
ᶠw_surface = level(ᶠw, half)
@. ᶠw_surface = Geometry.Covariant3Vector(Fields.level(ᶠinterp(Geometry.contravariant3(ᶜuₕ)), half) / Geometry.contravariant3(one(ᶠw_surface)))
```

Reply:
> 

Agreed that this changes now that we have u_bar and u_tilde defined following the 1364 merge - I think the above is true with the u_tilde^3 b.c. rather than the u_bar^3 bc

Reply:
yes, I meant u_tilde, not u_bar!